### PR TITLE
kubevela: init 1.9.8

### PR DESCRIPTION
**Pull request by superherointj**
kubevela: init 1.9.8

Release: https://github.com/kubevela/kubevela/releases/tag/v1.9.8

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by superherointj** [comment]
@bryanasdev000 Review?

File: pkgs/applications/networking/cluster/kubevela/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    repo = "kubevela";
+    rev = "v${version}";
+    hash = "sha256-vkq4XuSE+DeCMRBR7Z276TJIpOi0qEop7FBD2PPiQN4=";
+  };
+
+  vendorHash = "sha256-rtzPrbjIhxDUcOZOht0PKidCacvTYUNeNUbisgKKR2A=";
+
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/oam-dev/kubevela/version.VelaVersion=${version}"
```
Comment (bryanasdev000) [inline pkgs/applications/networking/cluster/kubevela/default.nix:26]: Maybe add "github.com/oam-dev/kubevela/version.GitRevision=undefined" like https://github.com/kubevela/kubevela/blob/62efa9c78744070aa3cbdf097d4440610f15ffb9/Dockerfile#L30 too?

File: pkgs/applications/networking/cluster/kubevela/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "An application delivery platform to deploy and operate applications in hybrid, multi-cloud environments";
+    downloadPage = "https://github.com/kubevela/kubevela";
+    homepage = "https://kubevela.io/";
+    license = lib.licenses.asl20;
+    maintainers = [ ];
```
Comment (bryanasdev000) [inline pkgs/applications/networking/cluster/kubevela/default.nix:61]: No maintainer?

**Review by bryanasdev000 — CHANGES_REQUESTED** [review]
Overall, LGTM, just a nitpick about ldflags and empty maintainers.

File: pkgs/applications/networking/cluster/kubevela/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/kubevela/default.nix:26]: First, thanks for the review.
I could add it but `GitRevision=undefined` does not seem to make a meaningful difference.

File: pkgs/applications/networking/cluster/kubevela/default.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/kubevela/default.nix:61]: I usually maintain packages [and many others that I'm not listed as a maintainer]. But I'm avoiding adding myself as a maintainer at this moment.

File: pkgs/applications/networking/cluster/kubevela/default.nix (hunk 1 above)
Comment (bryanasdev000) [reply pkgs/applications/networking/cluster/kubevela/default.nix:26]: Does not seems to have any code reference too.

File: pkgs/applications/networking/cluster/kubevela/default.nix (hunk 2 above)
Comment (bryanasdev000) [reply pkgs/applications/networking/cluster/kubevela/default.nix:61]: No problem for me.

**Review by bryanasdev000 — APPROVED** [review]

**Comment by bryanasdev000** [comment]
@superherointj could you please take a look on the tests?

https://github.com/NixOS/nixpkgs/pull/274721/checks?check_run_id=19706550471

**Comment by superherointj** [comment]
> @superherointj could you please take a look on the tests?
> 
> https://github.com/NixOS/nixpkgs/pull/274721/checks?check_run_id=19706550471

Fixed.

File: pkgs/applications/networking/cluster/kubevela/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+, lib
+, stdenv
+, testers
+, kubevela
+, nix-update-script
+}:
+
+buildGoModule rec {
+  pname = "kubevela";
+  version = "1.9.7";
```
Comment (bryanasdev000) [inline pkgs/applications/networking/cluster/kubevela/default.nix:13]: Sorry for the delay, but 1.9.8 is out.

https://github.com/kubevela/kubevela/releases/tag/v1.9.8

**Review by bryanasdev000 — CHANGES_REQUESTED** [review]
Diff LGTM, but there is a new release out, would be cool to init it on latest.

File: pkgs/applications/networking/cluster/kubevela/default.nix (hunk 3 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/kubevela/default.nix]: Updated.

**Review by bryanasdev000 — APPROVED** [review]
Diff and ofborg LGTM.

**Review by ulrikstrid — APPROVED** [review]
Builds and some basic commands runs

**Pull request merged**